File: backend/app/src/main/java/com/smash/domain/activity/FreePeriod.java
package com.smash.domain.activity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "free_period")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class FreePeriod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    @Builder
    public FreePeriod(LocalDate startDate, LocalDate endDate) {
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public void update(LocalDate startDate, LocalDate endDate) {
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public boolean contains(LocalDate date) {
        return !date.isBefore(startDate) && !date.isAfter(endDate);
    }
}

File: frontend/app/lib/features/free_period/model/free_period.dart
class FreePeriod {
  FreePeriod({
    required this.id,
    required this.name,
    required this.startDate,
    required this.endDate,
  });

  final int id;
  final String? name;
  final String startDate;
  final String endDate;

  factory FreePeriod.fromJson(Map<String, dynamic> json) {
    return FreePeriod(
      id: json['id'] as int,
      name: json['name'] as String?,
      startDate: json['startDate'] as String,
      endDate: json['endDate'] as String,
    );
  }
}
